gapic/src/main/com/google/gapid/views/CommandExperiments.java

```diff
@@ -0,0 +1,107 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.gapid.views;
+
+import com.google.gapid.models.Analytics.View;
+import com.google.gapid.models.Experiments;
+import com.google.gapid.models.Models;
+import com.google.gapid.proto.service.Service.ClientAction;
+import com.google.gapid.server.Client;
+import com.google.gapid.widgets.DialogBase;
+import com.google.gapid.widgets.Theme;
+import com.google.gapid.widgets.Widgets;
+
+import org.eclipse.jface.window.Window;
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.layout.GridData;
+import org.eclipse.swt.layout.GridLayout;
+import org.eclipse.swt.widgets.Button;
+import org.eclipse.swt.widgets.Composite;
+import org.eclipse.swt.widgets.Control;
+import org.eclipse.swt.widgets.Shell;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.logging.Logger;
+
+public class CommandExperiments {
+  private static final Logger LOG = Logger.getLogger(CommandEditor.class.getName());
+  private final Models models;
+  private final Theme theme;
+
+  public CommandExperiments(Client client, Models models, Theme theme) {
+    this.models = models;
+    this.theme = theme;
+  }
+
+  public static boolean shouldShowExperimentsPopup() {
+    return true;
+  }
+
+  public void showExperimentsPopup(Shell parent) {
+    models.analytics.postInteraction(View.Commands, ClientAction.ShowExperiments);
+    ExperimentsDialog dialog = new ExperimentsDialog(parent, models, theme);
+    if (dialog.open() == Window.OK) {
+      models.analytics.postInteraction(View.Commands, ClientAction.UpdateExperiments);
+      models.profile.setExperiments(dialog.newExperiments);
+      models.profile.reload();
+    }
+  }
+
+  public Experiments getExperiments() {
+    return this.models.profile.getExperiments();
+  }
+
+  public void SetDisableAF(boolean selected) {
+    models.analytics.postInteraction(View.Commands, ClientAction.DisableAF);
+    models.profile.getExperiments().setShouldDisableAF(selected);
+    models.profile.reload();
+  }
+
+  private static class ExperimentsDialog extends DialogBase {
+    private final Models models;
+    private final Map<String,Control> experimentControls = new HashMap<String,Control>();
+    public Experiments newExperiments;
+
+    public ExperimentsDialog(Shell parentShell, Models models, Theme theme) {
+      super(parentShell, theme);
+      this.models = models;
+    }
+
+    @Override
+    public String getTitle() {
+      return "Profile Experiments";
+    }
+
+    @Override
+    protected Control createDialogArea(Composite parent) {
+      Composite area = (Composite)super.createDialogArea(parent);
+      Composite container = Widgets.createComposite(area, new GridLayout(2, false));
+      container.setLayoutData(new GridData(SWT.FILL, SWT.FILL, true, true));
+      Widgets.createLabel(container, "Disable Anisotropic Filtering");
+      experimentControls.put("disableAF",
```

Comment:
Just keep a reference to the checkbox in this instance, there is no need to go via a map.